src/katsdpcontroller/schemas/__init__.py

```diff
@@ -2,48 +2,81 @@
 
 import codecs
 import json
-from distutils.version import StrictVersion
 
 import jinja2
 import jsonschema
 import pkg_resources
+from packaging.version import Version
 
 _env = jinja2.Environment(loader=jinja2.PackageLoader(__name__, "."))
 
 
+def _normalise_version(version):
+    """Coerce strings to ComparableVersion."""
+    return ComparableVersion(version) if isinstance(version, str) else version
+
+
 def _make_validator(schema):
     """Check a schema document and create a validator from it"""
     validator_cls = jsonschema.validators.validator_for(schema)
     validator_cls.check_schema(schema)
     return validator_cls(schema, format_checker=jsonschema.FormatChecker())
 
 
+class ComparableVersion(Version):
```

Comment:
How about using `functools.total_ordering`?

Reply:
I thought of that, but according to the docs it won't override comparators that are already implemented in a base class.